src/dynamichmc_inference.jl

```diff
@@ -8,19 +8,15 @@ end
 
 function (P::DynamicHMCPosterior)(a)
     @unpack alg, problem, likelihood, priors, kwargs = P
-
-    prob = problem_new_parameters(problem, a)
+    prob = remake(problem,u0 = convert.(eltype(a.a),problem.u0),p=a.a)
     sol = solve(prob, alg; kwargs...)
     if any((s.retcode != :Success for s in sol))
-        ℓ = -Inf
+        ℓ = LogDensityProblems.reject_logdensity()
```

Comment:
@tpapp is this a correct use of `reject_logdensity()`?

Reply:
Nope. Call it as a function, it will throw an exception that is caught by the wrappers and converted to `-Inf`. Or you can return `-Inf` directly, if you don't want to unwind the stack.

Reply:
Alright, then I'll just return `-Inf`